### PR TITLE
MgidX Bid Adapter : initial release

### PR DESCRIPTION
**Pull request by xmgiddev**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by Fawke** [comment]
@ChrisHuie Can you please help with the failing Circle Ci check? I ran the command multiple times but it's giving me some error [here](https://app.circleci.com/pipelines/github/prebid/Prebid.js/16100/workflows/098c76e2-06c9-4fe6-9c9a-bc685f8a7898/jobs/29544?invite=true#step-106-85)
Screenshot 2023-06-07 at 11 53 42 AM

It doesn't seem to be failing because of this PR, but the check is passing for other PRs currently open. Do we need to make another empty commit to make this pass? I remember you did something similar in the past to fix this.

**Comment by xmgiddev** [comment]
@Fawke oh, sorry, docs PR is here - https://github.com/prebid/prebid.github.io/pull/4610

File: modules/mgidXBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    const placements = [];
+    const request = {
+      deviceWidth,
+      deviceHeight,
+      language,
+      secure,
+      host,
+      page,
+      placements,
+      coppa: config.getConfig('coppa') === true ? 1 : 0,
```
Comment (patmmccann) [inline modules/mgidXBidAdapter.js:167]: please prefer bidder request object

File: modules/mgidXBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+      if (isPlainObject(gdprConsent) && typeof gdprConsent?.gdprApplies === 'boolean' && gdprConsent.gdprApplies) {
+        query.push('gdpr=1');
+      } else {
+        query.push('gdpr=0');
+      }
+      if (isPlainObject(uspConsent) && uspConsent?.consentString) {
+        query.push(`us_privacy=${encodeURIComponent(uspConsent?.consentString)}`);
+      }
+      if (isPlainObject(gppConsent) && gppConsent?.gppString) {
+        query.push(`gppString=${encodeURIComponent(gppConsent?.gppString)}`);
```
Comment (patmmccann) [inline modules/mgidXBidAdapter.js:230]: what about sid

Comment (xmgiddev) [reply modules/mgidXBidAdapter.js:230]: sorry, didn't notice that. now we don't need sid and gppString params

**Comment by ChrisHuie** [comment]
@xmgiddev pulling in commits currently on master should fix testing issue :)

**Review by Fawke — APPROVED** [review]

**Pull request merged**